api/src/main/java/org/apache/iceberg/actions/Cdc.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+public interface Cdc extends Action<Cdc, Cdc.Result> {
+  /**
+   * Emit changed data set by a snapshot id.
+   *
+   * @param snapshotId id of the snapshot to generate changed data
+   * @return this for method chaining
+   */
+  Cdc ofSnapshot(long snapshotId);
```

Comment:
If we decide to include a verb in the action name, I'd also consider renaming these methods like this:

```
// changes for a specific snapshot
GenerateChangelog forSnapshot(long snapshotId);

// changes starting from a particular snapshot (exclusive)
GenerateChangelog afterSnapshot(long fromSnapshotId);

// changes from start snapshot (exclusive) to end snapshot (inclusive)
GenerateChangelog betweenSnapshots(long fromSnapshotId, long toSnapshotId);
```